dbt/adapters/bigquery/retry.py

```diff
@@ -1,10 +1,9 @@
 from typing import Callable, Optional
 
-from google.api_core.exceptions import Forbidden
 from google.api_core.future.polling import DEFAULT_POLLING
 from google.api_core.retry import Retry
-from google.cloud.bigquery.retry import DEFAULT_RETRY
-from google.cloud.exceptions import BadGateway, BadRequest, ServerError
+from google.cloud.bigquery.retry import DEFAULT_RETRY, _job_should_retry
```

Comment:
While this is a private function, there is no reasonable way to capture BQ's defaults here. We need to add an exception into list that goes into this function, which would then require vendoring the whole function. Even if we remove `BadRequest`, we would then need to pull a private attribute (`_predicate`) off of `DEFAULT_JOB_RETRY` so that we can implement the retry cap (`job_retries`).